petab/parameters.py

```diff
@@ -0,0 +1,189 @@
+"""Functions operating on the PEtab parameter table"""
+
+import numbers
+import pandas as pd
+import numpy as np
+from collections import OrderedDict
+from typing import Iterable, Set, List
+
+import libsbml
+
+from . import lint, core, measurements, sbml
+
+
+def get_parameter_df(parameter_file_name: str) -> pd.DataFrame:
+    """
+    Read the provided parameter file into a `pandas.Dataframe`.
+    """
+
+    parameter_df = pd.read_csv(parameter_file_name, sep='\t')
+    lint.assert_no_leading_trailing_whitespace(
+        parameter_df.columns.values, "parameter")
+
+    try:
+        parameter_df.set_index(['parameterId'], inplace=True)
+    except KeyError:
+        raise KeyError(
+            'Parameter table missing mandatory field `parameterId`.')
+
+    return parameter_df
+
+
+def get_optimization_parameters(parameter_df: pd.DataFrame) -> List[str]:
+    """
+    Get list of optimization parameter ids from parameter dataframe.
+    """
+    return list(parameter_df.reset_index()['parameterId'])
+
+
+def create_parameter_df(sbml_model: libsbml.Model,
+                        condition_df: pd.DataFrame,
+                        measurement_df: pd.DataFrame,
+                        parameter_scale: str = 'log10',
+                        lower_bound: Iterable = None,
+                        upper_bound: Iterable = None) -> pd.DataFrame:
+    """Create a new PEtab parameter table
+
+    All table entries can be provided as string or list-like with length
+    matching the number of parameters
+
+    Arguments:
+        sbml_model: @type libsbml.Model
+        condition_df: @type pandas.DataFrame
+        measurement_df: @type pandas.DataFrame
+        parameter_scale: parameter scaling
+        lower_bound: lower bound for parameter value
+        upper_bound: upper bound for parameter value
+    """
+    parameter_ids = list(get_required_parameters_for_parameter_table(
+        sbml_model, condition_df, measurement_df))
+
+    df = pd.DataFrame(
+        data={
+            'parameterId': parameter_ids,
+            'parameterName': parameter_ids,
+            'parameterScale': parameter_scale,
+            'lowerBound': lower_bound,
+            'upperBound': upper_bound,
+            'nominalValue': np.nan,
+            'estimate': 1,
+            'priorType': '',
+            'priorParameters': ''
+        })
+    df.set_index(['parameterId'], inplace=True)
+
+    # For SBML model parameters set nominal values as defined in the model
+    for parameter_id in df.index:
+        try:
+            parameter = sbml_model.getParameter(parameter_id)
+            if parameter:
+                df.loc[parameter_id, 'nominalValue'] = parameter.getValue()
+        except ValueError:
+            # parameter was introduced as condition-specific override and
+            # is potentially not present in the model
```

Comment:
print this as a warning?

Reply:
There are many cases where this may happen, so warnings will be annoying. The proper checking is done in petab.lint.

Reply:
There are many cases where this may happen, so warnings will be annoying. The proper checking is done in petab.lint.